src/mysqlsh_helpers.py

```diff
@@ -275,18 +373,92 @@ def _run_mysqlcli_script(self, script: str, password=None) -> None:
         Args:
             script: raw SQL script string
             password: root password to use for the script when needed
+            user: user to run the script
         """
         command = [
             "/usr/bin/mysql",
             "-u",
-            "root",
+            user,
             "--protocol=SOCKET",
             f"--socket={MYSQLD_SOCK_FILE}",
             "-e",
             script,
         ]
         if password:
-            # passoword is needed after user
+            # password is needed after user
             command.append(f"--password={password}")
         process = self.container.exec(command)
         process.wait_output()
+
+    def _get_cluster_status(self) -> dict:
+        """Get the cluster status.
+
+        Executes script to retrieve cluster status.
+        Won't raise errors.
+
+        Returns:
+            Cluster status as a dictionary
+        """
+        status_commands = (
+            f"shell.connect('{self.cluster_admin_user}:{self.cluster_admin_password}@{self.instance_address}')",
+            f"cluster = dba.get_cluster('{self.cluster_name}')",
+            "print(cluster.status())",
+        )
+
+        try:
+            output = self._run_mysqlsh_script("\n".join(status_commands), verbose=0)
+            output_dict = json.loads(output.lower())
+            # pop topology from status due it being potentially too long
+            # and containing keys with `:` in it
+            output_dict["defaultreplicaset"].pop("topology")
+            return output_dict
+        except ExecError as e:
+            logger.exception(f"Failed to get cluster status for {self.cluster_name}", exc_info=e)
+
+    def _set_group_replication_initial_variables(self) -> None:
+        """Set group replication initial variables.
+
+        Necessary for k8s deployments.
+        Raises ExecError if the script gets a non-zero return code.
+        """
+        commands = (
+            "INSTALL PLUGIN group_replication SONAME 'group_replication.so';",
+            f"SET PERSIST group_replication_local_address='{self.instance_address}:33061';",
+            "SET PERSIST group_replication_ip_allowlist='0.0.0.0/0';",
+        )
+
+        self._run_mysqlcli_script(
+            " ".join(commands), self.cluster_admin_password, self.cluster_admin_user
+        )
+
+    def create_custom_config_file(self, report_host: str) -> None:
+        """Create custom configuration file.
+
+        Necessary for k8s deployments.
+        Raises ExecError if the script gets a non-zero return code.
```

Comment:
Change to `MySQLCreateCustomConfigFileError` (the error that is being raised).

Reply:
Hard catch that one, fixed on 79b9f97